src/Microsoft.NetCore.Analyzers/Core/Security/UseContainerLevelAccessPolicy.cs

```diff
@@ -0,0 +1,148 @@
+
+                    operationBlockStartContext.RegisterOperationAction(operationAnalysisContext =>
+                    {
+                        var invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
+                        var methodSymbol = invocationOperation.TargetMethod;
+
+                        if (methodSymbol.Name != "GetSharedAccessSignature")
+                        {
+                            return;
+                        }
+
+                        var namespaceSymbol = methodSymbol.ContainingNamespace;
+
+                        if (namespaceSymbol == null)
+                        {
+                            return;
+                        }
+
+                        var namespaceQualifiedName = namespaceSymbol.ToDisplayString();
+
+                        if (namespaceQualifiedName != "Microsoft.WindowsAzure.Storage" &&
+                            !namespaceQualifiedName.StartsWith("Microsoft.WindowsAzure.Storage.", StringComparison.Ordinal))
+                        {
+                            return;
+                        }
+
+                        var pointsToAnalysisResult = PointsToAnalysis.GetOrComputeResult(
+                                                                            invocationOperation.GetTopmostParentBlock().GetEnclosingControlFlowGraph(),
+                                                                            owningSymbol,
+                                                                            wellKnownTypeProvider,
+                                                                            interproceduralAnalysisConfig,
+                                                                            interproceduralAnalysisPredicateOpt: null,
+                                                                            false);
+
+                        foreach (var (nspace, policyIdentifierName) in NamespaceAndPolicyIdentifierNamePairs)
+                        {
+                            if (namespaceQualifiedName == "Microsoft.WindowsAzure.Storage." + nspace)
+                            {
+                                var argumentOperation = invocationOperation.Arguments.FirstOrDefault(s => s.Parameter.Name == policyIdentifierName);
+
+                                if (argumentOperation != null)
+                                {
+                                    var pointsToAbstractValue = pointsToAnalysisResult[argumentOperation.Kind, argumentOperation.Syntax];
+
+                                    if (pointsToAbstractValue.NullState == NullAbstractValue.Null)
+                                    {
+                                        operationAnalysisContext.ReportDiagnostic(
+                                            invocationOperation.CreateDiagnostic(
+                                                Rule));
+                                    }
+                                }
```

Comment:
>} [](start = 32, length = 1)

Do you also need to report when the method being invoked doesn't have the a policy identifier parameter?